_source/_posts/2022-08-12-ci-cd-microservices-circleci-spinnaker.md

```diff
@@ -0,0 +1,949 @@
+```
+
+Do the same updates to the `gateway/.circleci/config.yml` file. Additionally, the following change is required:
+
+- Add `~/.cache/Cypress` as a cache path
+
+Be sure to replace `your-dockerhub-username` and set the correct `IMAGE_NAME`.
+The final `config.yml` for the `gateway` must look like this:
+
+```yml
+version: 2.1
+jobs:
+  build:
+    environment:
+      DOCKERHUB_USER: your-dockerhub-username
+      IMAGE_NAME: your-dockerhub-username/gateway
+    machine:
+      image: ubuntu-2004:current
+    resource_class: large
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - {%raw%}v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+            # Perform a Partial Cache Restore (https://circleci.com/docs/2.0/caching/#restoring-cache)
+            - v1-dependencies-
+      - run:
+          name: Print Java Version
+          command: 'java -version'
+      - run:
+          name: Print Node Version
+          command: 'node -v'
+      - run:
+          name: Print NPM Version
+          command: 'npm -v'
+      - run:
+          name: Install Node Modules
+          command: 'npm install'
+      - save_cache:
+          paths:
+            - node
+            - node_modules
+            - ~/.gradle
+            - ~/.cache/Cypress
+          key: {%raw%}v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+      - run:
+          name: Give Executable Power
+          command: 'chmod +x gradlew'
+      - run:
+          name: Backend tests
+          command: npm run ci:backend:test
+      - run:
+          name: Run Front End Tests
+          command: npm run ci:frontend:test
+      - when:
+          condition:
+            equal: [ main, << pipeline.git.branch >> ]
+          steps:
+            - run:
+                name: Build and Publish Docker Image
+                command: |
+                  ./gradlew -Pprod bootJar jib \
+                    -Djib.to.image=$IMAGE_NAME \
+                    -Djib.to.auth.username=$DOCKERHUB_USER \
+                    -Djib.to.auth.password=$DOCKERHUB_PASS
+```
+
+To take advantage of the one-step GitHub integration of CircleCI, you need a [GitHub](https://github.com/signup) account. After signing in to GitHub, create a new public repository `store`. Follow the instructions to push the existing repository from your local machine to GitHub using the command line. Do the same with the `gateway` project.
+
+[Sign up](https://circleci.com/integrations/github) at CircleCI with your GitHub account, and on the left menu choose **Projects**. Find the `store` project and click **Set Up Project**. Configure the project to use the **Fastest** option (Use the `.circleci/config.yml` in my repo) and type `main` for the branch. Click **Set Up Project** to continue.
+
+{% img blog/jhipster-ci-cd/circleci-project.png alt:"CircleCI project setup form" width:"450" %}{: .center-image }
```

Comment:
It might look better if the black border is cropped from this image.

Reply:
Cropped.